### DISCUSSION TITLE
How to add custom comment provider?

### DISCUSSION THREAD

**Discussion by icewyvrn** [ppoffice/hexo-theme-icarus#1022] [Q&A] [answered]
I want to add [Cactus Comments](https://cactus.chat/docs/getting-started/introduction/) to Icarus theme, but I don't know where to start, where do I embed the comments? Thanks

**Comment by ppoffice** [comment, marked as answer]
@icewyvrn https://ppoffice.github.io/hexo-theme-icarus/uncategorized/faq/#Layout See **Where are the layout files for widgets/comments/share...? How do I customize built-in widgets/comments/share...?**
